### ISSUE TITLE
No Mic on ES8336

### ISSUE THREAD

**Issue by junocomp** [thesofproject/linux#3824]
Using the latest es8336 topologies and kernel 5.19, the microphone is not detected. Sound/speakers work fine.

**Comment by plbossart** [comment]
@junocomp kernel 5.19 is not good enough, please retry with the setup suggested here: https://github.com/thesofproject/linux/wiki/ES8336-support

**Comment by junocomp** [comment]
i tried it, still not detected.

**Comment by plbossart** [comment]
@junocomp try to see if jack detection works for you, see instructions in https://github.com/thesofproject/linux/issues/3800

**Comment by junocomp** [comment]
@plbossart When using a headset with mic, then is detected. But not the internal mic.

**Comment by plbossart** [comment]
There is no such thing as detecting an internal mic @junocomp. It's soldered at the hardware level, what you are missing it is the configuration to use it. It can be a mix of alsa-settings and/or GPIO settings.

**Comment by junocomp** [comment]
I still can't get the mic to work. I have the latest UCM and everything.

LSHW finds the mic but it wont enable it.

```
$ sudo lshw -class multimedia
[sudo] password for mobian: 
  *-multimedia:0 UNCLAIMED  
       description: Multimedia controller
       product: JasperLake IPU
       vendor: Intel Corporation
       physical id: 5
       bus info: pci@0000:00:05.0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: pciexpress msi pm cap_list
       configuration: latency=0
       resources: iomemory:600-5ff memory:6000000000-6000ffffff
*-multimedia:1
       description: Multimedia audio controller
       product: Jasper Lake HD Audio
       vendor: Intel Corporation
       physical id: 1f.3
       bus info: pci@0000:00:1f.3
       logical name: card0
       logical name: /dev/snd/controlC0
       logical name: /dev/snd/hwC0D2
       logical name: /dev/snd/pcmC0D0c
       logical name: /dev/snd/pcmC0D0p
       logical name: /dev/snd/pcmC0D5p
       logical name: /dev/snd/pcmC0D6p
       logical name: /dev/snd/pcmC0D7p
       version: 01
       width: 64 bits
       clock: 33MHz
       capabilities: pm msi bus_master cap_list
       configuration: driver=sof-audio-pci-intel-icl latency=32
       resources: iomemory:600-5ff iomemory:600-5ff irq:135 memory:6002120000-6002123fff memory:6002000000-60020fffff
```

**Comment by zheksoon** [comment]
Having the same story on KUU YoBook M (Gemini Lake N4020), no internal mic is detected, and `alsamixer` cannot enable the corresponding flag. But when I re-route MIC1 to headset using `SOC_ES8336_HEADSET_MIC1` quirk (`options snd-soc-sof-es8336 quirk=0x122`), I am able to capture internal microphone sound with `arecord -Dhw:0,0 -c2 -r48000 -d 10 -f S32_LE test.wav` (though of very bad quality with constant high-pitch hum)

**Comment by zheksoon** [comment]
Just checked again, without `SOC_ES8336_HEADSET_MIC1` quirk it's enough to switch Differential Mux to lin2-rin2 in `alsamixer` to capture internal microphone sound. So the problem is that no separate internal microphone device is available.

**Comment by junocomp** [comment]
@zheksoon Are you able to see your mic now on gnome-settings or is it picked up by any other apps?

Have you tried https://mictests.com/ ?

**Comment by zheksoon** [comment]
> @zheksoon Are you able to see your mic now on gnome-settings or is it picked up by any other apps?
> 
> Have you tried https://mictests.com/ ?

Nope, there is still no microphone device in pulse audio settings, so external applications don't recognize it's connected and pick up monitor device (which just replays output to the input)

**Comment by junocomp** [comment]
@zheksoon I tried your suggestion and I coudn't get the mic to detect at all, not even when create a wav file.

**Comment by zheksoon** [comment]
@junocomp both with the quirk and Differential Mux settings? Maybe try to play with different combinations in alsamixer, something should work for analog mic. And are you sure your mic isn't dmic?

**Comment by zheksoon** [comment]
@plbossart could you please elaborate on what should be done to make the internal microphone device to get detected and appear as input in Pulse Audio?

**Comment by plbossart** [comment]
There is no 'detection' of internal microphones. The only think that can interfere is the detection of the headset. If that is not working, then it's possible that you never see the internal mic.

One possibility is to comment out the 'conflicting device' in the UCM file, That should show you both inputs and give you the ability to select the relevant input.

**Comment by junocomp** [comment]
@zheksoon have you had any luck?

**Comment by zheksoon** [comment]
@junocomp nope, I tried to mess with ALSA UCM settings, but removing all conditions and conflict devices from it doesn't make a microphone to appear as a device. Anyway I had only sport interest in enabling it, its quality is so bad it will be unusable in real life.

**Comment by junocomp** [comment]
Any updates on this?

**Comment by junocomp** [comment]
No one?

**Comment by RezinoviyDed** [comment]
I got a new kernel (kernel 6.0.11-225.current), however, just like the author of the topic no microphones (dual mic) on Alldocube i1405 Jasper Lake N5100. When connect a 3.5mm mobile headset, the microphone does not work.

The speakers work good. The headphones work good.

**Comment by merura** [comment]
please also try the proaudio profile in pulseaudio config (for example with pulseaudio volume control GUI program)
and with the latest upstream kernel
